app/controllers/pallets_controller.rb

```diff
@@ -1,22 +1,43 @@
 class PalletsController < InheritedResources::Base
-  before_action :set_reception
+  before_action :set_reception , only: :create
+  before_action :storage_locations, only: :create
```

Comment:
storage_locations and product_location should not be declared in a callback because in the method set_location you will call they to assign in @location variable, actually this logic makes the request to database twice

Reply:
thank you, I thought about it at first but I was not sure